qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1092,38 +1092,81 @@ def update(self, md_template):
 
             # simple validations of sample ids and column names
             samples_diff = set(new_map.index).difference(current_map.index)
+            # Fail if we don't have the same samples
             if samples_diff:
                 raise QiitaDBError(
                     'The new template differs from what is stored '
                     'in database by these samples names: %s'
                     % ', '.join(samples_diff))
-            columns_diff = set(new_map.columns).difference(current_map.columns)
-            if columns_diff:
+
+            # Fail if we have new columns in the templates
+            # (only updates are allowed, not extensions)
+            if not set(current_map.columns).issuperset(new_map.columns):
```

Comment:
If this check is validating that the changed columns already exist in the original template, then I think the error message could be improved to be more descriptive.